src/appleseed/renderer/kernel/lighting/volumelightingintegrator.cpp

```diff
@@ -88,45 +84,57 @@ namespace
             else
                 emission_position = light_sample.m_point;
 
-            const Vector3d light_dir = emission_position - volume_ray.m_org;
+            // Direction from the ray origin to the light center.
+            const Vector3f origin_to_light(emission_position - volume_ray.m_org);
 
-            m_origin_to_center = dot(light_dir, volume_ray.m_dir);
+            // Signed distance from the ray origin
+            // to the light projection onto the ray.
+            m_origin_to_projection = dot(origin_to_light, Vector3f(volume_ray.m_dir));
 
-            m_center_to_light =
-                std::sqrt(square_norm(light_dir) - square(m_origin_to_center));
+            // Distance from the projection point to the light center (height).
+            m_projection_to_light =
+                std::sqrt(square_norm(origin_to_light) - square(m_origin_to_projection));
 
-            m_near_angle = std::atan2(-m_origin_to_center, m_center_to_light);
+            m_near_angle = std::atan2(-m_origin_to_projection, m_projection_to_light);
```

Comment:
Maybe you can simplify the math, isn't `m_near_angle` also simply `std::acos(m_origin_to_projection / std::norm(origin_to_light))`? You could compute both and check that they are always equal, in which case you could switch to that simpler formulation.